rest/persistent_config_test.go

```diff
@@ -1138,6 +1138,95 @@ func TestMigratev30PersistentConfig(t *testing.T) {
 	require.True(t, found)
 	require.Equal(t, "2-abc", migratedDb.Version)
 
+	// Verify legacy config has been removed
+	_, getError = sc.BootstrapContext.Connection.GetMetadataDocument(ctx, bucketName, PersistentConfigKey30(ctx, groupID), defaultDatabaseConfig)
+	require.Equal(t, base.ErrNotFound, getError)
+
+}
+
+func TestMigratev30PersistentConfigUseXattrStore(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("This test only works against Couchbase Server")
+	}
+
+	base.TestRequiresCollections(t)
+	base.SetUpTestLogging(t, base.LevelInfo, base.KeyHTTP, base.KeyConfig)
+
+	serverErr := make(chan error, 0)
+
+	// Set up test for persistent config
+	config := BootstrapStartupConfigForTest(t)
+	config.Unsupported.UseXattrConfig = base.BoolPtr(true)
+	// "disable" config polling for this test, to avoid non-deterministic test output based on polling times
+	config.Bootstrap.ConfigUpdateFrequency = base.NewConfigDuration(time.Minute * 10)
+	ctx := base.TestCtx(t)
+	sc, err := SetupServerContext(ctx, &config, true)
+	require.NoError(t, err)
+	defer func() {
+		sc.Close(ctx)
+		require.NoError(t, <-serverErr)
+	}()
+
+	go func() {
+		serverErr <- StartServer(ctx, &config, sc)
+	}()
+	require.NoError(t, sc.WaitForRESTAPIs(ctx))
+
+	// Get a test bucket, and use it to create the database.
+	tb := base.GetTestBucket(t)
+	defer func() {
+		fmt.Println("closing test bucket")
+		tb.Close(ctx)
+	}()
```

Comment:
I think you can use `RestTester` for this?

Reply:
This test is primarily doing bootstrap operations - what's the benefit of RestTester?

Reply:
The benefit would be that there is potentially less boilerplate and consistent ways of starting a `ServerContext` and closing the server context and it would open a bucket for you.

I'm not set on this and you have to be able to pass `BootstrapOptions` into `RestTester`. I have seen this pattern before where it is used because we didn't have support in `RestTester` for the given functionality.

Reply:
I agree with making that change, but would prefer to minimize the changes in this PR, to manage the backport impact (the other 8 usages of this pattern in this file should also be changed).     Filed https://issues.couchbase.com/browse/CBG-3498 for this enhancement.